packages/curve-ui-kit/src/features/connect-wallet/lib/wagmi/useWagmi.tsx

```diff
@@ -0,0 +1,63 @@
+import { useCallback, useMemo } from 'react'
+import { useAccount, useClient, useConnect, useDisconnect, useEnsName } from 'wagmi'
+import { useGlobalState } from '@ui-kit/hooks/useGlobalState'
+import type { Wallet } from '../types'
+import { createWallet } from './adapter'
+import { connectors } from './connectors'
+import { supportedWallets, type WalletType } from './wallets'
+
+export const useWalletType = () => useGlobalState<WalletType, null>('wallet', null)
+export const useConnectPromise = () =>
+  useGlobalState<[resolve: (wallet: Wallet) => void, reject: (err: unknown) => void], null>('wagmiConnectWallet', null)
+
+export const useWagmi = () => {
+  // when the modal is displayed, we save a promise to resolve later - this is for compatibility with existing code
+  const [connecting, setConnecting] = useConnectPromise()
+
+  // this is the wallet type selected in the modal
+  const [walletType, setWalletType] = useWalletType()
```

Comment:
What's the purpose of storing this with `useGlobalState`?

Reply:
The point was to make sure the wallet is returning the right label in every hook instance
https://github.com/curvefi/curve-frontend/blob/782f37afe89e0fb9c97eb949a1d6f3e15006cb1c/packages/curve-ui-kit/src/features/connect-wallet/lib/wagmi/useWagmi.tsx#L57-L60